flutter-idea/src/io/flutter/run/LaunchState.java

```diff
@@ -156,14 +158,31 @@ protected RunContentDescriptor launch(@NotNull ExecutionEnvironment env) throws
     else {
       descriptor = new RunContentBuilder(result, env).showRunContent(env.getContentToReuse());
     }
+    final String nameWithDeviceName = descriptor.getDisplayName() + " (" + device.deviceName() + ")";
+    boolean displayNameUpdated = false;
     try {
-      final Field f = descriptor.getClass().getDeclaredField("myDisplayName");
+      final Field f = descriptor.getClass().getDeclaredField("myDisplayNameView");
```

Comment:
Could you add a comment (somewhere) that explains why we have to do all this introspection? (And why we need to look for `"myDisplayNameView"` and fall back to `"myDisplayName"`? (Your PR description might be sufficient.)

Reply:
Sure! Updated

Reply:
Super. Thank you!